include/aws/s3/s3_client.h

```diff
@@ -10,6 +10,7 @@
 #include <aws/s3/s3.h>
 
 #include <aws/auth/signing_config.h>
+#include "aws/io/uri.h"
```

Comment:
Trivial: You don't need to include this at header file. Using forward declaration, you can move this include to `s3_client.c` instead

Reply:
ok